### DISCUSSION TITLE
How to filter JSON to only take the latest entry by date?

### DISCUSSION THREAD

**Discussion by deltamelter** [grafana/grafana-infinity-datasource#233] [Q&A]
Have JSON containing an array of historical prices, I want infinity datasource to take only the latest by datetime.
How can I do that with this plugin?
I have 4 queries feeding a custom panel.  Or do I have to code this in the panel?

alternatively, is there a way to sort by say date, so that the top result is the latest.

**Comment by yesoreyeram** [comment]
In case if your data is already sorted, some Grafana panels can already do that. ( stats panel )

**Reply by deltamelter** [reply]
Unfortunately it's not sorted and needs filtering on name as well as sorting by date.
I maybe could do it in the custom panel but I don't really know typescript or how to sort and filter using it.

**Reply by yesoreyeram** [reply]
Have u tried 0.8 version of plugin which supports UQL. That may help

**Reply by deltamelter** [reply]
no, still using latest released.  I'll try on your demo grafana...
